enhancements/sig-architecture/6-placements/README.md

```diff
@@ -0,0 +1,517 @@
+- Create metrics data when creating/updating placement decisions (user story 9). It needs further investigation and should be covered in a separate enhancement;
+- Support time window (user story 10). It should be covered in a separate enhancement;
+
+## Proposal
+
+### API Spec
+
+#### Placement API
+`Placement` is a namespace scoped API and represents a selection of clusters in one or multiple ManagedClusterSets. When a placement is created, a controller consuming the placement should generate one or multiple placement decisions based on the spec defined in placement.
+```go
+// Placement defines a rule to select a set of ManagedClusters from the ManagedClusterSets bound
+// to the placement namespace.
+//
+// Here is how the placement policy combines with other selection methods to determine a matching
+// list of clusters:
+// 1) Clusters are registered with hub as cluster-scoped ManagedClusters;
+// 2) ManagedClusters are organized into cluster-scoped ManagedClusterSets;
+// 3) ManagedClusterSets are bound to workload namespaces;
+// 4) Namespace-scoped Placements specify a slice of ManagedClusterSets which select a working set
+//    of potential ManagedClusters;
+// 5) Then Placements subselect from that working set using name/label/claim selection.
+//
+// No ManagedClusters will be selected if there is no ManagedClusterSet bound to the placement
+// namespace. User is able to bind a ManagedClusterSet to a namespace by creating a 
+// ManagedClusterSetBinding in the namespace if they have an RBAC rule to CREATE on the virtual
+// subresource of `managedclustersets/bind`.
+//
+// A slice of PlacementDecisions with label placement.open-cluster-management.io={placement name}
+// will be created to represent the cluster selected by this placement.
+// 
+// If a ManagedCluster is selected and added into the PlacementDecisions, other components may apply
+// workload on it; once it is removed from the PlacementDecisions, the workload applied on this
+// ManagedCluster should be evicted accordingly.
+type Placement struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of Placement.
+	Spec PlacementSpec `json:"spec"`
+
+	// Status represents the current status of the Placement
+	// +optional
+	Status PlacementStatus `json:"status,omitempty"`
+}
+
+// PlacementSpec defines the attributes of Placement.
+// An empty PlacementSpec selects all clusters from the bound ManagedClusterSets to placement namespace.
+type PlacementSpec struct {
+	// Predicates represent a slice of predicates to select clusters. The predicates are ORed.
+	// +optional
+	Predicates []ClusterPredicate `json:"predicates,omitempty"`
+
+	// Affinity represents the scheduling constraints
+	// +optional
+	Affinity *Affinity `json:"affinity,omitempty"`
+}
+
+// ClusterPredicate represents a predicate to select clusters.
+type ClusterPredicate struct {
```

Comment:
if the type is an union, better having a discriminator field to fit k8s union api structure https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1027-api-unions

Reply:
Thanks for the suggestion. I removed `ClusterName` field from `ClusterPredicate` because user can get cluster name from both cluster labels and cluster claims. Now this struct is no longer a union type.

Reply:
You may have already noticed that a placement will select no cluster if there is no clusterset bound to the placement namespace. So a cluster belongs to no clusterset will never been selected by any placement. This is one of the limitations of the current design. We'll figure out a way to fix it in beta phase.

> ### Beta
>- The new APIs is able to cover all the capabilities of the current placementrule APIs;
>- Conversion webhook is provided to migrate existing placement objects to the new placement APIs;-
>- At lease one component uses the new APIs;
>- **Support selecting clusters belong to no ManagedClusterSet;**
>- Support tolerations/taints (user story 8);
>- Create metrics data when creating/updating placement decisions (user story 9);
>- Support time window (user story 10);

In addition, without ClusterName field, it is still possible to select clusters with names from bound clustersets by using either label (see example below) or claim (see example 1 in proposal).
```
apiVersion: cluster.open-cluster-management.io/v1alpha1
kind: Placement
metadata:
  name: clusters-with-specific-names
  namespace: ns1
spec:
  predicates:
    - requiredDuringSchedulingRequiredDuringExecution:
        labelSelector:
          matchExpressions:
            - key: name
              operator: In
              values:
                - cluster1
                - cluster2
```